test/test_poisson_solvers.jl

```diff
@@ -64,3 +65,35 @@ function poisson_ppn_planned_div_free_cpu(ft, Nx, Ny, Nz, planner_flag)
 
     ∇²ϕ.data ≈ RHS_orig.data
 end
+
+"""
+    Test that the Poisson solver can recover an analytic solution. In this test, we are trying to see if the solver
+    can recover the solution ``\\Psi(x, y, z) = cos(\\pi m_z z / L_z) sin(2\\pi m_y y / L_y) sin(2\\pi m_x x / L_x)``
+    by giving it the source term or right hand side (RHS), which is ``f(x, y, z) = \\nabla^2 \\Psi(x, y, z) =
+    -((\\pi m_z / L_z)^2 + (2\\pi m_y / L_y)^2 + (2\\pi m_x/L_x)^2) \\Psi(x, y, z)``.
+"""
+function poisson_ppn_recover_sine_cosine_solution(ft, Nx, Ny, Nz, Lx, Ly, Lz, mx, my, mz)
```

Comment:
This is just semantics, but I don't like that the ordinary action of the Poisson solver is called 'recovery'... perhaps just call this `poisson_solver_test_sinusoid` or something?

Reply:
That's definitely a better name. Will update.